modules/apps/portal-store/portal-store-gcs/src/main/java/com/liferay/portal/store/gcs/GCSStore.java

```diff
@@ -386,59 +407,37 @@ private void _initEncryption() {
 	}
 
 	private void _initGCSStore() throws PortalException {
-		String serviceAccountKey = _gcsStoreConfiguration.serviceAccountKey();
-
-		if (Validator.isNotNull(serviceAccountKey)) {
-			try (InputStream inputStream = new ByteArrayInputStream(
-					serviceAccountKey.getBytes())) {
-
-				_googleCredentials = ServiceAccountCredentials.fromStream(
-					inputStream);
-			}
-			catch (IOException ioException) {
-				throw new PortalException(
-					"Unable to authenticate with GCS", ioException);
-			}
-		}
-
-		StorageOptions.Builder storageOptionsBuilder =
-			StorageOptions.newBuilder(
-			).setRetrySettings(
-				RetrySettings.newBuilder(
-				).setInitialRetryDelay(
-					Duration.ofMillis(
-						_gcsStoreConfiguration.initialRetryDelay())
-				).setInitialRpcTimeout(
-					Duration.ofMillis(
-						_gcsStoreConfiguration.initialRPCTimeout())
-				).setJittered(
-					_gcsStoreConfiguration.retryJitter()
-				).setMaxAttempts(
-					_gcsStoreConfiguration.maxRetryAttempts()
-				).setMaxRetryDelay(
-					Duration.ofMillis(_gcsStoreConfiguration.maxRetryDelay())
-				).setMaxRpcTimeout(
-					Duration.ofMillis(_gcsStoreConfiguration.maxRPCTimeout())
-				).setRetryDelayMultiplier(
-					_gcsStoreConfiguration.retryDelayMultiplier()
-				).setRpcTimeoutMultiplier(
-					_gcsStoreConfiguration.rpcTimeoutMultiplier()
-				).build()
-			);
-
-		if (_googleCredentials != null) {
-			storageOptionsBuilder.setCredentials(_googleCredentials);
+		try {
+			_googleCredentials = _getGoogleCredentials();
```

Comment:
No need to extract a separate method here. Please inline.

Reply:
Thank you for your feedback, @dantewang! The PR has be re-sent as https://github.com/liferay-core-infra/liferay-portal/pull/2153.